### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by firmusoft**
- Collect All Test Files for Coverage.
- More Unit Tests.

### CODE REVIEW

File: src/UICommand.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    const obj = uiCmd.dryRunEditorStateProxyGetter(editor.state, 'tr');
+    expect(obj).toBeInstanceOf(Transaction);
+
+    if (obj instanceof Transaction) {
+      const meta = obj.getMeta('dryrun');
+      expect(meta).toEqual(true);
+    }
+  });
+
+  it('dryRunGet Any', () => {
```
Comment (melgish) [inline src/UICommand.test.js:72]: Use test language to describe the branch being tested, and test (at minimum) every branch that will be encountered under normal circumstances.

```javascript
describe('dryRunEditorStateProxyGetter', () => {
    let tr: Transaction;
    let state;
    let uiCmd: UICommand;

    beforeEach(() => {
      tr = new Transaction({});
      tr.setMeta = jest.fn().mockImplementation(() => tr);
      // use same spy for both cases
      state = {tr, other: tr};
      uiCmd = new UICommand();
    });

    describe('when getting the transaction"', () => {
      it('should update transaction metadata', () => {
        const output = uiCmd.dryRunEditorStateProxyGetter(state, 'tr');

        expect(tr.setMeta).toHaveBeenCalled();
        expect(output).toBe(state.tr);
      });
    });

    describe('when getting other data', () => {
      it('should not update transaction metadata', () => {
        const output = uiCmd.dryRunEditorStateProxyGetter(state, 'other');

        expect(tr.setMeta).not.toHaveBeenCalled();
        expect(output).toBe(state.other);
      });
    });
  });
```

**Review by melgish — CHANGES_REQUESTED** [review]
Tests should be created to verify that the code works as intended.  Good tests are descriptive and tell anyone looking at the code what the behavior being tested is.   Test written for the sole purpose of increasing code coverage are a waste of time.

File: src/UICommand.test.js [hunk 2]
```diff
@@ -57,4 +57,26 @@ describe('UICommand', () => {
     const enabled = uiCmd.isEnabled(editor.state, editor.view);
     expect(enabled).toEqual(false);
   });
+
+  it('dryRunGet Transaction', () => {
```
Comment (melgish) [inline src/UICommand.test.js:61]: Please follow the format described in previous comments.

File: src/UICommand.test.js [hunk 3]
```diff
@@ -62,6 +62,11 @@ describe('UICommand', () => {
     const uiCmd = new UICommand();
     const obj = uiCmd.dryRunEditorStateProxyGetter(editor.state, 'tr');
     expect(obj).toBeInstanceOf(Transaction);
+
+    if (obj instanceof Transaction) {
```
Comment (melgish) [inline src/UICommand.test.js:66]: Generally tests should not branch.  During execution line 64 will abort the test if it does not pass.  Therefore you will never get to line 66 if it's not a Transaction

File: package.json [hunk 4]
```diff
@@ -96,7 +96,6 @@
     ]
   },
   "jest": {
-    "rootDir": "src",
```
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply package.json:99]: Got you. Shall keep jest.config.js and move the required to jest.config.js.

File: src/UICommand.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    // because you can `unmock` it in clean way doing `mockRestore`
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterAll(() => {
+    // Restore mock after all tests are done, so it won't affect other test suites
+    console.error.mockRestore();
+  });
+
+  it('isEnabled', () => {
```
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js:20]: Sure will make corrections accordingly. Thanks!

File: src/UICommand.test.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    const uiCmd = new UICommand();
+    const enabled = uiCmd.isEnabled(editor.state, editor.view);
+    expect(enabled).toEqual(false);
+  });
+
+  it('shouldRespondToUIEvent', () => {
```
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js:26]: Will make sure to follow this. Thanks!

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply package.json:99]: Combined and removed additional jest.config.js

File: src/UICommand.test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js]: Corrected

File: src/UICommand.test.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js]: Corrected

File: src/index.test.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  const editor = createEditor(doc(p('<cursor>')));
+  const KEY = 'uniqueKey';
+  const VAL = 'uniqueValue';
+
+  it('should construct', () => {
+    const step = new SetDocAttrStep(KEY, VAL);
+    expect(step).toBeTruthy();
+  });
+
+  it('should register', () => {});
```
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/index.test.js]: Added expectation

File: src/UICommand.test.js (hunk 1 above)
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js]: Modified and Used this. Thanks!

File: src/index.test.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  it('should toJSON', () => {
+    const sdaStep = new SetDocAttrStep(KEY, VAL);
+    expect(sdaStep.toJSON()).toStrictEqual({
+      stepType: 'SetDocAttr',
+      key: KEY,
+      value: VAL,
+    });
+  });
+
+  xit('should register only once', () => {
```
[the remark this answers is not in the document]
Comment (ashfaq-shamsudeen) [reply src/index.test.js]: Added comments.

File: src/UICommand.test.js (hunk 3 above)
Comment (ashfaq-shamsudeen) [reply src/UICommand.test.js:92]: Need to correct this test and some others... So skipping those tests for now. Have put comments accordingly.

**Pull request merged**